lib/media-controls-impl.js

```diff
@@ -0,0 +1,524 @@
+
+  // If we get a touchstart, don't process mouse events
+  if (e.type === 'touchstart') {
+    this.shadowRoot.removeEventListener('mousedown', this);
+    this.shadowRoot.removeEventListener('mousemove', this);
+    this.shadowRoot.removeEventListener('mouseup', this);
+  }
+
+  switch(e.type) {
+
+    case 'mousedown':
+      //
+      // The component is listening to window 'mousemove' events so
+      // that the slider movement will function even when the mouse
+      // moves off the play head. However, if the component is always
+      // listening to the window events, it would receive very many
+      // spurious 'mousemove' events. To prevent this, the component
+      // only listens to 'mousemove' events after receiving a 'mousedown'
+      // event.
+      //
+      if (!this.mouseEventHandlerRegistered) {
+        window.addEventListener('mousemove', this, true);
+        window.addEventListener('mouseup', this, true);
+        this.mouseEventHandlerRegistered = true;
+      }
+
+      // fall through to touchstart...
+
+    case 'touchstart':
+      switch(e.target) {
+        case this.els.play:
+          //
+          // Let the 'play' and 'pause' handlers take care of changing
+          // the icon and setting the l10n-id (for the screen reader).
+          //
+          if (this.mediaPlayer.paused) {
+            this.mediaPlayer.play();
+          }
+          else {
+            this.mediaPlayer.pause();
+          }
+          break;
+
+        case this.els.seekForward:
+        case this.els.seekBackward:
+
+          var direction = null;
+          if (e.target === this.els.seekForward) {
+            direction = 1;
+          } else if (e.target === this.els.seekBackward) {
+            direction = -1;
+          } else {
+            return;
+          }
+
+          var offset = direction * 10;
+          this.seekBy(this.mediaPlayer.currentTime + offset);
+
+          // Begin the "longpress" movement after a one second delay.
+          var self = this;
+          this.intervalId = window.setInterval(function() {
+              self.seekBy(self.mediaPlayer.currentTime + offset);
+            }, 1000);
+          break;
+      }
+      break;
+
+      case 'touchend':
+      case 'mouseup':
+        //
+        // If ending a long-press forward or backward, clear timer
+        //
+        if (this.intervalId) {
+           window.clearInterval(this.intervalId);
+           this.intervalId = null;
+        }
+        else if (this.dragging) {
+          // If ending a movement of the slider, end slider movement
+          this.handleSliderMoveEnd();
+        }
+
+        if (e.type === 'mouseup') {
+          // Don't listen for mousemove and mouseup until we get a mousedown
+          window.removeEventListener('mousemove', this, true);
+          window.removeEventListener('mouseup', this, true);
+          this.mouseEventHandlerRegistered = false;
+        }
+        break;
+
+    case 'loadedmetadata':
```

Comment:
From the music app, I've learned that there is also a durationchange (or durationchanged?) event. It might sometimes be fired for variable bit rate media when the duration actually does change. So you should probably register a listener and handle it here exactly as you handle loadedmetadata